### ISSUE TITLE
Trouble with install on Windows

### ISSUE THREAD

**Issue by dmcglinn** [MoBiodiv/MoBspatial#3]
I'm having a different problem than in #2 on my Windows machine. It appears that the .dll file is causing some trouble - I don't think you typically include pre-compiled files in packages so I guess not knowing any of the details I would suggest dropping the file `MoBspatial.dll` from the package. But again I'm really not sure.

```
 R.version
               _                           
platform       x86_64-w64-mingw32          
arch           x86_64                      
os             mingw32                     
system         x86_64, mingw32             
status                                     
major          3                           
minor          3.1                         
year           2016                        
month          06                          
day            21                          
svn rev        70800                       
language       R                           
version.string R version 3.3.1 (2016-06-21)
nickname       Bug in Your Hair         
```
```R
install_github('MoBiodiv/MoBspatial')
Downloading GitHub repo MoBiodiv/MoBspatial@master
from URL https://api.github.com/repos/MoBiodiv/MoBspatial/zipball/master
Installing MoBspatial
"C:/PROGRA~1/R/R-33~1.1/bin/x64/R" --no-site-file  \
  --no-environ --no-save --no-restore --quiet CMD  \
  INSTALL  \
  "C:/Users/Daniel/AppData/Local/Temp/RtmpCmydMy/devtools172c22b255bc/MoBiodiv-MoBspatial-4bd41e1"  \
  --library="C:/Users/Daniel/Documents/R/win-library/3.3"  \
  --install-tests 

* installing *source* package 'MoBspatial' ...
** libs

*** arch - i386
make: Nothing to be done for `all'.
installing to C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/i386

*** arch - x64
make: Nothing to be done for `all'.
installing to C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/x64
** R
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
Error in inDL(x, as.logical(local), as.logical(now), ...) : 
  unable to load shared object 'C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/i386/MoBspatial.dll':
  LoadLibrary failure:  %1 is not a valid Win32 application.

Error: loading failed
Execution halted
*** arch - x64
ERROR: loading failed for 'i386'
* removing 'C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial'
Error: Command failed (1)
```

**Comment by FelixMay** [comment]
Hi Dan,
On windows you need to have installed the Rtools for Windows, but I guess you are aware of that? I will check out the code on my machine later today …
Did the error just occur on the Windows machine or also on other platforms?

Should I remove the dll from the GitHUb repository?

Felix

Von: Dan McGlinn [mailto:notifications@github.com]
Gesendet: Montag, 12. September 2016 04:11
An: MoBiodiv/MoBspatial <MoBspatial@noreply.github.com>
Betreff: [MoBiodiv/MoBspatial] Trouble with install on Windows (#3)


I'm having a different problem than in #2<https://github.com/MoBiodiv/MoBspatial/issues/2> on my Windows machine. It appears that the .dll file is causing some trouble - I don't think you typically include pre-compiled files in packages so I guess not knowing any of the details I would suggest dropping the file MoBspatial.dll from the package. But again I'm really not sure.

 R.version

               _

platform       x86_64-w64-mingw32

arch           x86_64

os             mingw32

system         x86_64, mingw32

status

major          3

minor          3.1

year           2016

month          06

day            21

svn rev        70800

language       R

version.string R version 3.3.1 (2016-06-21)

nickname       Bug in Your Hair

install_github('MoBiodiv/MoBspatial')

Downloading GitHub repo MoBiodiv/MoBspatial@master

from URL https://api.github.com/repos/MoBiodiv/MoBspatial/zipball/master

Installing MoBspatial

"C:/PROGRA~1/R/R-33~1.1/bin/x64/R" --no-site-file  \

  --no-environ --no-save --no-restore --quiet CMD  \

  INSTALL  \

  "C:/Users/Daniel/AppData/Local/Temp/RtmpCmydMy/devtools172c22b255bc/MoBiodiv-MoBspatial-4bd41e1"  \

  --library="C:/Users/Daniel/Documents/R/win-library/3.3"  \

  --install-tests



* installing *source* package 'MoBspatial' ...

** libs



*** arch - i386

make: Nothing to be done for `all'.

installing to C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/i386



*** arch - x64

make: Nothing to be done for `all'.

installing to C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/x64

** R

** preparing package for lazy loading

** help

*** installing help indices

** building package indices

** installing vignettes

** testing if installed package can be loaded

*** arch - i386

Error in inDL(x, as.logical(local), as.logical(now), ...) :

  unable to load shared object 'C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial/libs/i386/MoBspatial.dll':

  LoadLibrary failure:  %1 is not a valid Win32 application.



Error: loading failed

Execution halted

*** arch - x64

ERROR: loading failed for 'i386'

* removing 'C:/Users/Daniel/Documents/R/win-library/3.3/MoBspatial'

Error: Command failed (1)

**Comment by dmcglinn** [comment]
Before you do anything to the repo are you able to recreate my error?

**Comment by dmcglinn** [comment]
Oh and I do have Rtools installed 3.1.1

**Comment by FelixMay** [comment]
Hi Dan,
Yes, I also git your error. I did not notice it because before I always built my package locally. And I checked in Wickham’s R packages page that there should be indeed no compiled code on GitHub. So I removed all compiled files (object files and dll files) from the repo. Now it works on my machine with install_github(). So thanks for pointing on this issue! And let me know if ot works for you also now. Still I recommend that you update your Rtools and Rcpp version.

Let me know when you are far enough with the function documentation and I can start the vignette with a step by step example. Is there any fixes need from my side within the boxplot functions?

Felix

Von: Dan McGlinn [mailto:notifications@github.com]
Gesendet: Montag, 12. September 2016 14:29
An: MoBiodiv/MoBspatial <MoBspatial@noreply.github.com>
Cc: May, Felix <felix.may@idiv.de>; Comment <comment@noreply.github.com>
Betreff: Re: [MoBiodiv/MoBspatial] Trouble with install on Windows (#3)


Oh and I do have Rtools installed 3.1.1
